### PR TITLE
PAYARA-3209 Added vendor support for MP-JWT namespaced claims

### PR DESCRIPTION
**Pull request by Pandrex247**
Original PR: https://github.com/payara/Payara/pull/3328

### CODE REVIEW

**Comment by Pandrex247** [comment]
Jenkins test please

**Comment by arjantijms** [comment]
Needs running MP-JWT TCK

File: appserver/payara-appserver-modules/microprofile/jwt-auth/src/main/java/fish/payara/microprofile/jwtauth/jwt/JwtTokenParser.java [hunk 1]
```diff
@@ -63,9 +64,19 @@
...
     private final List<Claims> requiredClaims = asList(iss, sub, exp, iat, jti, groups);
+    
+    private final boolean enableNamespacedClaims;
+    private final Optional<String> customNamespace;
 
+    public JwtTokenParser(Optional<Boolean> enableNamespacedClaims, Optional<String> customNamespace) {
+        this.enableNamespacedClaims = enableNamespacedClaims.orElse(false);
+        this.customNamespace = customNamespace;
+    }
+
```
Comment (MattGill98) [inline appserver/payara-appserver-modules/microprofile/jwt-auth/src/main/java/fish/payara/microprofile/jwtauth/jwt/JwtTokenParser.java]:
```suggestion

    public JwtTokenParser() {
        this(Optional.empty(), Optional.empty());
    }
```

Our TCK runner depends on the default constructor.

**Comment by MattGill98** [comment]
jenkins test please

**Pull request merged**